src/main/java/org/broadinstitute/consent/http/service/DACAutomationRuleService.java

```diff
@@ -115,48 +113,74 @@ public void triggerDACRuleSettings(User researcher, List<Integer> datasetIds, St
         }
       });
     });
 
     if (!datasetsAuthorized.isEmpty()) {
-      try {
-        emailService.sendDACAutomationApprovalResearcherMessage(researcher, datasetsAuthorized.stream()
-            .map(d -> new DatasetMailDTO(d.getName(), d.getDatasetIdentifier()))
-            .toList(), dar.getDarCode(), datasetsAuthorized.stream()
-            .map(dataset -> useRestrictionConverter.translateDataUse(dataset.getDataUse(), DataUseTranslationType.DATASET))
-            .distinct()
-            .collect(Collectors.joining(";")));
-      } catch (Exception e) {
-        logWarn(e.getMessage());
-        logWarn(e.getCause().getMessage());
-        throw new InternalServerErrorException(
-            "Error while sending Dac Automation messages.", e);
-      }
+      sendEmail(researcher, datasetsAuthorized, dar);
     }
   }
+
+  @VisibleForTesting
+  protected void sendEmail(User researcher, List<Dataset> datasetsAuthorized, DataAccessRequest dar) {
+    try {
+      emailService.sendDACAutomationApprovalResearcherMessage(researcher, datasetsAuthorized.stream()
+          .map(d -> new DatasetMailDTO(d.getName(), d.getDatasetIdentifier()))
+          .toList(), dar.getDarCode(), datasetsAuthorized.stream()
+          .map(dataset -> useRestrictionConverter.translateDataUse(dataset.getDataUse(), DataUseTranslationType.DATASET))
+          .distinct()
+          .collect(Collectors.joining(";")));
+    } catch (Exception e) {
+      logWarn(e.getMessage());
+      logWarn(e.getCause().getMessage());
+      throw new InternalServerErrorException(
+          "Error while sending Dac Automation messages.", e);
+    }
+  }
+
+  @VisibleForTesting
+  protected void applyRule(DACAutomationRule rule, Dataset dataset, DataAccessRequest dar,
+      List<Dataset> datasetsAuthorized) {
+    RuleImplementationInterface ruleImplementation = getRuleImplementation(rule);
+    boolean shouldApprove = ruleImplementation.compare(dataset, dar);
+    if (shouldApprove) {
+      openElectionAndApprove(rule, ruleImplementation, dar, datasetsAuthorized, dataset);
+    } else {
+      logInfo(String.format("Rule %s not triggered for DAC id: %s and dataset id: %s", rule.ruleType(), dataset.getDacId(),
+          dataset.getDatasetId()));
+    }
+  }
+
+  @VisibleForTesting
+  protected static RuleImplementationInterface getRuleImplementation(
+      DACAutomationRule rule) {
+    DACAutomationRuleType type = rule.ruleType();
+    return Rules.implementationList.stream()
+        .filter(r -> r.getRuleType().equals(type))
+        .findFirst()
+        .orElseThrow(() -> new IllegalArgumentException(
+            String.format("No rule implementation found for type: %s", type)));
+  }
+
+  @VisibleForTesting
+  protected void openElectionAndApprove(DACAutomationRule rule, RuleImplementationInterface ruleImplementation,
+      DataAccessRequest dar, List<Dataset> datasetsAuthorized, Dataset dataset) {
+    // TODO _ Check on new election type ...
```

Comment:
I think we can remove the TODO on the election type because we've decided that the juice isn't worth the squeeze with making a custom election.  The vote type implementation below memorializes the vote being performed on behalf of DACBOT/RADAR.